### PR TITLE
Move WDA to appium-webdriveragent

### PR DESCRIPTION
**Pull request by imurchie**
Get rid of the crufty submodule and start importing the `appium-webdriveragent` package. At the moment that package just gives the WDA code and the `checkForDependencies` function. There is much else that can (and should) be moved out of here and into there, but it can wait.

### CODE REVIEW

File: lib/wda/xcodebuild.js [hunk 1]
```diff
@@ -152,7 +154,6 @@ class XcodeBuild {
     // first do a build phase
```
Comment (mykola-mokhnach) [inline lib/wda/xcodebuild.js]: the previous condition can be deleted, since Xcode 7 is not supported anyway

File: lib/wda/xcodebuild.js [hunk 2]
```diff
@@ -152,7 +154,6 @@ class XcodeBuild {
...
     log.debug('Pre-building WDA before launching test');
     this.usePrebuiltWDA = true;
-    this.xcodebuild = await this.createSubProcess(true);
```
Comment (mykola-mokhnach) [inline lib/wda/xcodebuild.js]: is it intended?

Comment (imurchie) [reply lib/wda/xcodebuild.js]: Yes, I noticed while going through the logs to fix a problem, that we created the process here, but then recreated it when we ran start. This one is just thrown away.

File: lib/wda/xcodebuild.js (hunk 1 above)
Comment (imurchie) [reply lib/wda/xcodebuild.js]: That should be part of a different PR.

**Pull request merged**